### ISSUE TITLE
Exllama support

### ISSUE THREAD

**Issue by chaseaucoin** [outlines-dev/outlines#338]
I searched and didn't see this already requested. 

### Presentation of the new feature

Exllama is a language implementation that uses hand crafted cuda kernels to dramatically speed up inference of quantized large language models. 

I use it in production because it provides the most effective use of hardware. 

I would like to have outlines work with exllama. 

There was a blog article of an individual trying to do this with guidance

https://betterprogramming.pub/prototyping-a-small-guidance-clone-with-exllama-logits-bias-de26f85623ab

### Where does it fit in Outlines?

Currently none of the other libraries based on "Efficient Guided Generation for Large Language Models" are taking advantage of exllama.  I think given the more straightforward nature of outlines under the hood it has the highest likelihood of being compatible and would give it a competitive advantage from a perspective of token/sec throughput making it more valuable in production scenarios. 

### Are you willing to open a PR?

I am but I might need a bit of guidance on the right point of abstraction in outlines to take advantage of. I believe an adapter can be made that reimplements ExLlamaV2Sampler to take into consideration a bias that is generated by the outlines generator.

**Comment by rlouf** [comment]
This looks relevant to the library. I'll need to take a deeper look at exllama to be able to tell you how feasible that would be. An integration would need to implement the same interface as [Transformers](https://github.com/outlines-dev/outlines/blob/3667e04b57feed66faa82fa64eee84421c6eb9e0/outlines/models/transformers.py#L59).

**Comment by rlouf** [comment]
All you need to do is to wrap [the `forward` method](https://github.com/turboderp/exllamav2/blob/1e6bebf0cad5355e7f2b15df4437fba61907deb3/exllamav2/model.py#L509) and init of an `exLlamaV2` model the same way we did for `transformers` models [here](https://github.com/outlines-dev/outlines/blob/027b29b45a752e360fdc038fe3fa0db0648fa0e0/outlines/models/transformers.py#L58).
